symengine/expression.h

```diff
@@ -173,11 +173,6 @@ inline Expression expand(const Expression &arg)
     return expand(arg.get_basic());
 }
 
-inline bool unified_eq(const Expression &a, const Expression &b)
```

Comment:
I was getting an ambiguous function call, in calling `unified_eq(umap_basic_num, umap_basic_num)`. So I moved the `unified_eq` to [this](https://github.com/symengine/symengine/pull/1003/files#diff-2b0bba2525514ed902f57cc760843b54R136)

I think compiler could not figure out which function to call? The expression one or the templatized one?